tests/models/test_utils.py

```diff
@@ -570,3 +572,19 @@ def test_flatten_nested_params():
     }
 
     assert _flatten_nested_params(rag_params) == expected_rag_flattened_params
+
+
+@pytest.mark.parametrize(
+    ("data", "target"),
+    [
+        (pd.DataFrame([{"a": [1, 2, 3]}]), [{"a": [1, 2, 3]}]),
+        (pd.DataFrame([{"a": np.array([1, 2, 3])}]), [{"a": [1, 2, 3]}]),
+        (pd.DataFrame([{0: np.array(["abc"])[0]}]), ["abc"]),
+        (np.array([1, 2, 3]), [1, 2, 3]),
+        (np.array([123])[0], 123),
+    ],
+)
+def test_convert_llm_input_data(data, target):
+    result = _convert_llm_input_data(data)
+    assert result == target
+    assert not isinstance(result, np.generic)
```

Comment:
What's the expected type?

Reply:
python generic types

Reply:
```suggestion
    assert isinstance(result, ???)
```

can we use `assert isinstance` instead?

Reply:
```
>>> import numpy as np
>>> x = np.array(["a"])[0]
>>> x
'a'
>>> isinstance(x, str)
True
>>> type(x)
<class 'numpy.str_'>
>>> isinstance(x, np.generic)
True
```
In such case, x is instance of `str` but it's also `np.generic`

Reply:
Is `"a"` also `np.generic`?

Reply:
<class 'numpy.str_'> is np.generic, but pure python string "a" is not

Reply:
Can we use this?

```python
@pytest.mark.parametrize(
    ("data", "target", "expected_type"),
    [
        ...
        (np.array([123])[0], 123, int),
    ],
)
def test_convert_llm_input_data(data, target):
    result = _convert_llm_input_data(data)
    assert result == target
    assert type(result) is expected_type?
```

Reply:
Updated!